node/src/proxy_server/mod.rs

```diff
@@ -851,10 +979,77 @@ pub trait IBCDHelper {
         msg: InboundClientData,
         retire_stream_key: bool,
     ) -> Result<(), String>;
+
+    fn request_route_and_transmit(
+        &self,
+        tth_args: TryTransmitToHopperArgs,
+        route_source: Recipient<RouteQueryMessage>,
+        proxy_server_sub: Recipient<RouteResultMessage>,
+    );
+}
+
+trait RouteQueryResponseResolver: Send {
+    fn resolve_message(
+        &self,
+        args: TryTransmitToHopperArgs,
+        proxy_server_sub: Recipient<RouteResultMessage>,
+        route_result: Result<Option<RouteQueryResponse>, MailboxError>,
```

Comment:
The type of this parameter is more complicated than its name would imply. I'm not yet sure why it's a `Result`, but at least it seems as though the name should end with `_opt`.